### PR TITLE
serve micro batch server with gunicorn

### PR DESCRIPTION
**Pull request by hrmthw**
(Thanks for sending a pull request! Please make sure to read the contribution guidelines, then fill out the blanks below.)

What changes were proposed in this pull request?
------------------------------------------------

Does this close any currently open issues?
------------------------------------------

How was this patch tested?
--------------------------

### CODE REVIEW

File: bentoml/cli/__init__.py [hunk 1]
```diff
@@ -36,7 +36,7 @@
 from bentoml.cli.aws_sagemaker import get_aws_sagemaker_sub_command
 from bentoml.cli.bento import add_bento_sub_command
 from bentoml.server import BentoAPIServer, get_docs
-from bentoml.server.marshal_server import MarshalServer
+from bentoml.server.marshal_server import MarshalServer, GunicornMarshalServer
```
Comment (parano) [inline bentoml/cli/__init__.py]: MarshalServer no longer in use?

File: bentoml/cli/__init__.py [hunk 2]
```diff
@@ -325,18 +325,19 @@ def serve_gunicorn(
...
-            )
             with reserve_free_port() as api_server_port:
-                marshal_server = MarshalServer(
-                    port=port, target_host="localhost", target_port=api_server_port
+                marshal_server = GunicornMarshalServer(
+                    target_host="localhost",
+                    target_port=api_server_port,
+                    bundle_path=bento_service_bundle_path,
+                    port=port,
+                    num_of_workers=1,
```
Comment (parano) [inline bentoml/cli/__init__.py]: should we make this configurable? so it's easier for us to do experiments in benchmark later

File: bentoml/server/marshal_server.py [hunk 3]
```diff
@@ -69,3 +72,65 @@ def async_start(self):
...
+    def load(self):
+        server = MarshalServer(self.target_host, self.target_port, port=self.port)
+        bento_service_metadata_pb = load_bento_service_metadata(
+            self.bento_service_bundle_path
+        )
+        server.setup_routes_from_pb(bento_service_metadata_pb)
+        return server.marshal_app.make_app()
+
+    def run(self):
+        track_server('gunicorn', {"number_of_workers": self.cfg.workers})
```
Comment (parano) [inline bentoml/server/marshal_server.py]: let's separate this from gunicorn server in telemetrics:
```
        track_server('gunicorn-marshal', {"number_of_workers": self.cfg.workers})
```

File: bentoml/cli/__init__.py (hunk 1 above)
Comment (hrmthw) [reply bentoml/cli/__init__.py]: No, using in `bento serve`

File: bentoml/cli/__init__.py [hunk 4]
```diff
@@ -299,8 +299,20 @@ def serve(port, bento=None, with_conda=False, enable_microbatch=False):
         default=False,
         help="(Alpha) Run API server with micro batch enabled",
     )
+    @click.option(
+        '--microbatch-workers',
+        type=click.INT,
+        default=1,
+        help="(Alpha) Number of microbatch workers",
```
Comment (parano) [inline bentoml/cli/__init__.py]: this seems a bit confusing, but let's revisit it before the release

File: bentoml/cli/__init__.py [hunk 5]
```diff
@@ -299,8 +299,20 @@ def serve(port, bento=None, with_conda=False, enable_microbatch=False):
...
+    )
     def serve_gunicorn(
-        port, workers, timeout, bento=None, with_conda=False, enable_microbatch=False
+        port,
+        workers,
+        timeout,
+        bento=None,
+        with_conda=False,
+        enable_microbatch=False,
+        microbatch_workers=1,
```
Comment (parano) [inline bentoml/cli/__init__.py]: this still needs to be passed to the with_conda code block

**Pull request merged**